### PR TITLE
Fixes #6.

### LINKED ISSUE #6: setup nose tests

### PR DESCRIPTION
**Pull request by Velrok**
Also included two files with comments as a test structure proposition.

### CODE REVIEW

File: spec/test_live_intetration.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"""
+This are blackbox integration test agains the wobble api using
```
Comment (zeisss) [inline spec/test_live_intetration.py:2]: against

File: spec/test_live_intetration.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+the WobbleService class.
+It purpose is to ensure that the WobbleService is capable of
+calling the wobble endpoint and return valid data.
+
+No internals should be test. Also keep the dependencies agains
```
Comment (zeisss) [inline spec/test_live_intetration.py:7]: against

File: spec/test_live_intetration.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+return data assumptions to a minimun. Test for structure not
+content.
+"""
+from nose.tools import *
+from wobble import *
+
+WOBBLE_API_ENDPOINT = "http://wobble.moinz.de/endpoint.php"
```
Comment (zeisss) [inline spec/test_live_intetration.py:14]: I would love to see these tests not beeing run against the LIVE system.

File: spec/test_wobble_service.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+"""
+WobbeService whitebox tests.
+This test should test complex internal processes.
+Only test really complex logic that would result in
+confusing errors in whitebox interation testing.
```
Comment (zeisss) [inline spec/test_wobble_service.py:5]: interaction? Integration?

**Comment by Uepsilon** [comment]
@ZeissS is right. Furthermore, testing against another project can mean that errors do not occure due to incorrect implementation on our side but changes in the api on the server-side. We should create a mockup for our implementation so that we can be sure that our code works as expected.

for serverside-check we can use the version for a first check, later on we can think about some code to harden it against changes.

**Comment by Velrok** [comment]
All valid points if we would talk about a big system.
But for now I don't think that test will be run by anyone else then the 3 of us.

@ZeissS there is no stage to test against ;) .  

@Uepsilon of course they will break. Thats the idea behind the black box tests. Testing agains a mock will not test the integration.

I am fully aware that spamming the live system with test request is ( in general ) bad. BUT I think the load is and will be negligible.  

Am Sonntag, 17. März 2013 | KW 11 um 16:04 schrieb Uepsilon:

**Comment by Uepsilon** [comment]
I'm still not comfortable with the idea of testing against the server and not against our mockup. we will not be able to tell whether our implementation has is incorrect or the api changed.

**Comment by Velrok** [comment]
The white box test should cover the stuff that is going on inside the class.
We can and should introduce a mock there.

Breaking integration test just tell you that the library is not working. Indeed reasons may be manyfold.
But again. The main idea would be that this test just tell you that using the lib one can communicate with the live system.
Think of if as the group truth. If thouse test fail we have to fix something! This situation is better than having all the stuff work because we have mocked everything, but in the wild it is failing because the api changed.

As @ZeissS told me: there will be no backward compatebility insurance.

Am Sonntag, 17. März 2013 | KW 11 um 17:21 schrieb Uepsilon:

**Pull request closed without merging**